### DISCUSSION TITLE
[Feature] API Scan External Library sub path or single file

### DISCUSSION THREAD

**Discussion by hanpq** [immich-app/immich#7712] [Feature Request]
### The feature

First, thank you all for the amazing work on Immich!

I utilize an automated processes for managing photos throughout different apps and mainly use Immich for consumption of already tagged, structured photos which in turn means I rely exclusively on external libraries. These libraries are stored on bind-mounted CIFS shares so I will never be able to use native file-watchers in Immich.

I would like to have the ability to tell Immich that a file has been added or updated in the external library and request Immich to add it or update it in case it already exists, rather than rescanning the whole library for modified files.

Would it be possible to add an option for the /library/:id/scan API to scan only a specific subfolder within the defined external library path and/or a single file?

Thanks again

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by shivanshs9** [comment]
+1 on this. For certain use cases, where I use a script to automate fixing EXIF metadata with date/geolocation, I would like to refresh that on Immich database as well.
With a API similar to [scanLibrary](https://api.immich.app/endpoints/libraries/scanLibrary), but exclusively for a specific list of paths, it'll be very fast to update metadata anytime.

**Comment by donnchawp** [comment]
It would be great to have an API endpoint like this. I use a script to move files into the external library, and because I have a huge external library, the watcher was taking up 3GB of RAM. Once I disabled watching on external libraries my RAM usage declined significantly.

Now, when I add new photos or videos I enqueue a `LibrarySyncFiles` job to scan those particular files only. It's an awful hack that I'll have to maintain.

**Comment by vposham** [comment]
First of all, thank you for making Immich. 

+1 on this. 

Running Immich against a nas hosted external library (phone photos land in Synology first, Immich scans the same NAS path). Currently on hourly full scans just to keep sync lag reasonable, but the whole library gets rescanned every time even though realistically (mostly) only the most recent folder/date range ever has new content.

Concretely, something like a two-tier schedule per external library would solve this well:

**Full scan**: infrequent (e.g. every 6-24h) which catches anything structural, deletions, moves, metadata drift.

**Targeted scan**: frequent (e.g. every 15-60min), configurable per library to a specific subfolder or date/month pattern (many photo tools, including Synology Photos itself, organize by `YYYY/MM`), so a fast, cheap scan only touches the current period's folder rather than walking the entire tree.

This would let people running a phone→NAS→Immich pipeline (rather than uploading directly through the app) get near-real-time sync without paying full-tree I/O cost on every cycle. Would also make it practical to trigger a scan immediately after a known write event (e.g. a Synology Photos sync completing) rather than relying purely on a blind interval.
